docs/reference/remap/functions/log.cue

```diff
@@ -27,15 +27,21 @@ remap: functions: log: {
 			}
 			default: "info"
 		},
+		{
+			name:        "rate_limit_secs"
+			description: "Specifies that the log message is output no more than once per the given number of seconds."
+			required:    false
+			default:     1
```

Comment:
I'm not entirely sure we should have a default here. It could be confusing for users to have log messages that they have specifically written themselves be rate-limited for no obvious reason. We do want to balance that against the likelihood of them flooding their own logs, but when they're writing these logs themselves it seems a little heavy-handed to do this for them (and it doesn't seem like they can opt out?).

Reply:
Agreed :+1:

Reply:
Safety first guys. I think this will be a very common foot-gun if we don't protect against this. At the very least we should require the user to choose. For this specific context, I am in favor of a default. The downside of not setting a default seems much worse than the downside of setting one. The log messages should make it clear that rate limiting is happening.

For example, this will be a common pattern:

```ruby
parsed, err = parse_syslog(.message)
if err {
  log("Syslog parsing failed: " + err)
} else {
  # ... proceed with processing
}
```

If Vector gets flooded with messages that fail parsing it'll bring Vector a halt without rate limiting. I think this will be a common scenario, any upstream error could trigger this.